pkg/ddl/notifier/store.go

```diff
@@ -87,17 +88,28 @@ func (t *tableStore) UpdateProcessed(
 	se *sess.Session,
 	ddlJobID int64,
 	multiSchemaChangeID int64,
-	processedBy uint64,
+	oldProcessedBy uint64,
+	newProcessedBy uint64,
 ) error {
 	sql := fmt.Sprintf(`
 		UPDATE %s.%s
```

Comment:
I’m not sure under the repeatable read isolation (snapshot isolation) whether TiDB guarantees that we can get the current processed_by_flag instead of the snapshot value.

Reply:
> get the current processed_by_flag 

I'm not clear about your meaning. `oldProcessedBy` is read from another SQL session (`sess4List`), and here the processing SQL session uses WHERE processed_by_flag = `oldProcessedBy` and affected rows to check if the value has been changed

Reply:
Oh, I see, they are in a different session.

Reply:
Thank you!